app/src/main/java/com/android/shelfLife/ui/newoverview/NewHouseHoldCreationScreen.kt

```diff
@@ -0,0 +1,283 @@
+package com.android.shelfLife.ui.newoverview
+
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.outlined.Close
+import androidx.compose.material.icons.outlined.Delete
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.shelfLife.R
+import com.android.shelfLife.model.newCreationScreen.CreationScreenViewModel
+import com.android.shelfLife.model.newInvitations.InvitationRepository
+import com.android.shelfLife.model.newhousehold.HouseHoldRepository
+import com.android.shelfLife.model.user.UserRepository
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.ui.navigation.Screen
+import com.android.shelfLife.ui.newutils.DeletionConfirmationPopUp
+import com.android.shelfLife.ui.utils.CustomButtons
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun HouseHoldCreationScreen(
```

Comment:
Have a look at my branch for the edit and add food item screens because I feel like there's a lot of logic and variables in this class that should be in the view model. This is not consistent with what people like me, Ricardo and Alejandro have been doing for view models and UI. It could be that we are the ones in the wrong but i think we still need to discuss it anyway.

Reply:
I have a lot of trouble understanding the logic, will have to find the person who wrote it.